src/shark/utilityconsole/commands/rkz/FindWeapons.java

```diff
@@ -0,0 +1,758 @@
+                break;
+
+            default:
+                Console.showMessage("Unsupported criteria encountered in constructParameterCriteria! Received " + criteria);
+        }
+        ParameterCriterion.CriteriaParameter cp = new ParameterCriterion.CriteriaParameter(criteria, data);
+        retVal = new ParameterCriterion(cp, criteriaQuantity);
+
+        return retVal;
+    }
+
+    public boolean containsWeaponSizes(String fullInputString) {
+        boolean retVal = false;
+        retVal = fullInputString.toLowerCase().contains("small")
+                || fullInputString.toLowerCase().contains("medium")
+                || fullInputString.toLowerCase().contains("large");
+
+        return retVal;
+    }
+
+    public String replaceWeaponSizesWithOrdinals(String fullInputString) {
+        // We don't really care about the position of the shipsize word,
+        // just replace all of them in one go, it'll hit what it hits, and call it done.
+        return fullInputString.toLowerCase()
+                .replaceAll("small", String.valueOf(WeaponAPI.WeaponSize.SMALL.ordinal()))
+                .replaceAll("medium", String.valueOf(WeaponAPI.WeaponSize.MEDIUM.ordinal()))
+                .replaceAll("large", String.valueOf(WeaponAPI.WeaponSize.LARGE.ordinal()));
+    }
+
+    public String replaceWeaponSizeWithExpression(String fullInputString) {
+        // While we generally don't care about the position of the "just the shipsize word",
+        // this time we want to replace it with an expression that limits to just that particular size
+        return fullInputString.toLowerCase()
+                .replaceAll("small", "size = " + WeaponAPI.WeaponSize.SMALL.ordinal())
+                .replaceAll("medium", "size = " + WeaponAPI.WeaponSize.MEDIUM.ordinal())
+                .replaceAll("large", "size = " + WeaponAPI.WeaponSize.LARGE.ordinal());
+    }
+
+    public boolean containsWeaponTypeExpectingKeyword(String fullInputString) {
+        // I believe only 'type = ' and 'mounttype = ' should expect a weapon type
+        boolean retVal = false;
+        retVal = fullInputString.toLowerCase().contains("type")
+                || fullInputString.toLowerCase().contains("mounttype")
+                || fullInputString.toLowerCase().contains("mount-type")
+                || fullInputString.toLowerCase().contains("mount_type");
+
+        return retVal;
+    }
+
+    public int countWeaponTypeExpectingKeywords(String fullInputString) {
+        int retVal = 0;
+        if (containsWeaponTypeExpectingKeyword(fullInputString)) {
+//            retVal += countOccurrences(fullInputString.toLowerCase(), "mounttype");
+//            retVal += countOccurrences(fullInputString.toLowerCase(), "mount-type");
+//            retVal += countOccurrences(fullInputString.toLowerCase(), "mount_type");
+//            retVal += countOccurrences(fullInputString.toLowerCase(), "type");
```

Comment:
hmmm.... deleting this will eventually leading to repeating this mistake.
not deleting it will just leave this garbage in the code...

🤔 🤔 🤔

Reply:
i'm leaving it. i forget things. it's better if i forget why i didn't delete it than "refactor dumb-self code" and break it.

Reply:
resolving. won't do.